arangod/Graph/Graph.cpp

```diff
@@ -157,7 +160,12 @@ Graph::Graph(TRI_vocbase_t& vocbase, std::string&& graphName,
       _vertexColls(),
       _edgeColls(),
       _numberOfShards(1),
-      _replicationFactor(vocbase.replicationFactor()),
+      _replicationFactor(vocbase.getDatabaseConfiguration().isOneShardDB
+                             ? (std::max)(vocbase.replicationFactor(),
+                                          vocbase.server()
+                                              .getFeature<ClusterFeature>()
+                                              .systemReplicationFactor())
+                             : vocbase.replicationFactor()),
```

Comment:
How does this come to pass? Do we have to do this everywhere we want to get the actual replication factor?

Reply:
No we do not have to.
In general we need to set distributeShardsLike for OneShard, and everything will unfold automatically.

For Graphs however we store the replicationFactor as part of the Graph Configuration, and use this value when creating the collection (in addition to oneShard distributeShardsLike) this is only to make those aligned and thereby not have our correctness checker throw an error.